### PR TITLE
Fix stream source lookup and add in optional External to StreamSource

### PR DESCRIPTION
**Pull request by derekcollison**
Signed-off-by: Derek Collison <derek@nats.io>

/cc @nats-io/core

### CODE REVIEW

File: server/stream.go [hunk 1]
```diff
@@ -117,7 +118,7 @@ type StreamSource struct {
 	FilterSubject string          `json:"filter_subject,omitempty"`
 	External      *ExternalStream `json:"external,omitempty"`
 
-	// Internal
+	// Internale
```
Comment (kozlovic) [inline server/stream.go:121]: "Internale": why the change?

**Review by kozlovic — APPROVED** [review]
LGTM (except for typo maybe?)

File: server/stream.go (hunk 1 above)
Comment (derekcollison) [reply server/stream.go:121]: Spelling?

**Review by matthiashanel — COMMENTED** [review]
LGTM
